src/ucp/proto/proto_common.inl

```diff
@@ -371,4 +368,79 @@ ucp_proto_request_pack_rkey(ucp_request_t *req, ucp_md_map_t md_map,
     return packed_rkey_size;
 }
 
+static UCS_F_ALWAYS_INLINE size_t
+ucp_proto_request_pack_iov_rkeys(ucp_request_t *req, ucp_md_map_t md_map,
+                                 uint64_t distance_dev_map,
+                                 const ucs_sys_dev_distance_t *dev_distance,
+                                 void *rkey_buffer)
+{
+    const ucp_datatype_iter_t *dt_iter = &req->send.state.dt_iter;
+    ucp_mem_h *memh                    = dt_iter->type.iov.memh;
+    void *p                            = rkey_buffer;
+    size_t iov_count, iov_index;
+    size_t packed_count, *ppacked_count, *ppacked_rkey_size;
+    ssize_t packed_rkey_size;
+    size_t total_size;
+
+    ucs_assert(dt_iter->dt_class == UCP_DATATYPE_IOV);
```

Comment:
pls use assertv and print the real class

Reply:
done